### PR TITLE
Release version 4.2.0

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Release version 4.2.0 - GA4 comming baby!!!
| Type?             | 
| BC breaks?        | 
| Deprecations?     | 
| Fixed ticket?     | 
| Sponsor company   | 
| How to test?      |

### CODE REVIEW

**Review by kpodemski — APPROVED** [review]

**Comment by Hlavtox** [comment]
One last change needs to be done, we should not be sending data to all gtag tags on the page. I will add `event_to`.

**Comment by Hlavtox** [comment]
One last change should probably be done before release. If user is using another gtag on the page, is it a problem that we are sending all of those events also to Adwords for example? 

It could be easily solved by adding `send_to: 'G-XXXXX'` to the events.

WDYT? @rmilecki @alejoasotelo @jf-viguier @okom3pom @Progi1984

**Comment by kpodemski** [comment]
@Hlavtox, there's no such thing as Adwords anymore 😝 I've checked the docs, etc., and it seems it is indeed better to add `send_to`.

**Comment by Hlavtox** [comment]
@kpodemski Ready to roll out 🎸

**Review by kpodemski — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Review by hibatallahAouadni — COMMENTED** [review]
Hello @Hlavtox 

There seems to be a problem with https://tagassistant.google.com/ I can't access any of my stores (even though some of them are hosted), am I missing something?

[Tag Assistant [Connecting].webm](https://user-images.githubusercontent.com/23308427/223433326-89bd0b18-3afe-4fb7-adc9-ff9e7cd39cf5.webm)


Thanks!

**Comment by kpodemski** [comment]
@hibatallahAouadni you sure you've checked hosted versions that are not behind maintenance? otherwise, it's impossible to check them with these tools

**Comment by Hlavtox** [comment]
I am using this on all my stores and it works perfectly, you are doing something wrong.

**Review by hibatallahAouadni — APPROVED** [review]
Hello @Hlavtox 

I don't know what exactly happened yesterday but it works now 🎉 
LGTM, QA ✅ 

[pr_131.webm](https://user-images.githubusercontent.com/23308427/223633517-3cbcbfd0-0389-4938-a5c4-9da37234d1e4.webm)

Thanks!

**Comment by nafribs** [comment]
i posted this in another place not sure if these issues are fixed..

Hey guys i been testing this module , and i found few issues..

in google analytics.. under Monetization - > Ecommerce purchases .Item revenue column is showing 0 . while Items purchased has data.

i am not sure if this related but the purchase event is not being imported into google ads. i have linked the event in the conversion settings in google ads.

**Comment by MrThrust** [comment]
Hi Guys... I've been eagerly awaiting the release... thanks for all the great work!.. 
we installed across our sites this morning, the only issue we found was on 1 site.. 
an older version of prestashop 1.6.1.1 

We received this error during checkout step 5 "payment". 
Unfortunately this older server still runs PHP - 5.6.40 ( _which, I suspect is the main issue_ )

`Catchable fatal error: Argument 1 passed to PrestaShop\Module\Ps_Googleanalytics\Hooks\HookActionCarrierProcess::getGoogleAnalytics4() must be an instance of PrestaShop\Module\Ps_Googleanalytics\Hooks\string, string given, called in /home/siteroot/sitename/modules/ps_googleanalytics/classes/Hook/HookActionCarrierProcess.php on line 58 and defined in /home/siteroot/sitename/modules/ps_googleanalytics/classes/Hook/HookActionCarrierProcess.php on line 79`

File: views/templates/hook/ps_googleanalytics.tpl [hunk 1]
```diff
@@ -17,39 +17,57 @@
...
+  <script async src="https://www.googletagmanager.com/gtag/js?id={$gaAccountId|escape:'htmlall':'UTF-8'}"></script>
+  <script>
+    window.dataLayer = window.dataLayer || [];
+    {literal}function gtag(){dataLayer.push(arguments);}{/literal}
+    gtag('js', new Date());
+    gtag(
+      'config',
+      '{$gaAccountId|escape:'htmlall':'UTF-8'}',
+      {ldelim}
+        'debug_mode':true
```
Comment (jf-viguier) [inline views/templates/hook/ps_googleanalytics.tpl:30]: debug mode should not be hardcoded to true. It can be activated with a chrome extension.

**Comment by nizelg** [comment]
> We received this error during checkout step 5 "payment". Unfortunately this older server still runs PHP - 5.6.40 ( _which, I suspect is the main issue_ )
Same on PHP 8.0 Prestashop 8.0.1 release

**Pull request merged**